libs/dyn/convert/to_typed.go

```diff
@@ -282,6 +282,11 @@ func toTypedFloat(dst reflect.Value, src dyn.Value) error {
 }
 
 func toTypedInterface(dst reflect.Value, src dyn.Value) error {
+	if src.Kind() == dyn.KindNil {
+		dst.Set(reflect.Zero(dst.Type()))
+		return nil
+	}
+
```

Comment:
Please also add unit test coverage in this package.

Reply:
done